lib/writer/block-write-stream.js

```diff
@@ -134,10 +157,24 @@ class BlockWriteStream extends stream.Writable {
     }
 
     fs.write(this.fd, chunk, 0, chunk.length, chunk.position, (error, bytesWritten) => {
-      this.bytesWritten += bytesWritten
-      this.delta += bytesWritten
-      this.blocksWritten += 1
-      this.position += bytesWritten
+      if (!error) {
+        this.bytesWritten += bytesWritten
+        this.delta += bytesWritten
+        this.blocksWritten += 1
+        this.position += bytesWritten
+        this.retries = 0
+        next(error)
+        return
+      }
+
+      const isTransient = BlockWriteStream.isTransientError(error)
+
+      if (isTransient && (this.retries < this.maxRetries)) {
+        this.retries += 1
+        this._write(chunk, encoding, next)
```

Comment:
Does "something else" also need to be done here, given that every call to `_write` does `this.bytesRead += chunk.length` ?

Reply:
It only does that if there hasn't been an error

Reply:
Are you sure? It's possible I'm not understanding the code properly, but from what I can see:
 *  the `_write (chunk, encoding, next) {` function starts on line 130
 * on line 141 you then do `this.bytesRead += chunk.length`
 * on line 159 you do the actual `fs.write(...`
 * on line 161 you do `this.bytesWritten += bytesWritten` if there wasn't an error
 * if the error was transient, you then do `this._write(chunk, encoding, next)` on line 174

So AFAICT if you get a bunch of transient errors then `this.bytesWritten` will be correct, but `this.bytesRead` will be incremented too much?

Reply:
Oh, sorry I misread `bytesRead` – yes, you're right.

Reply:
Fixed!